R/drive_list.R

```diff
@@ -80,6 +80,18 @@ drive_list <- function(path = NULL, pattern = NULL, ..., verbose = TRUE){
 get_leafmost_id <- function(path) {
 
   path_pieces <- unlist(strsplit(path, "/"))
+
+  root_id <- root_folder() ## store so we don't make repeated API calls
+
+  if (all(path_pieces == "~")) {
+    return(root_id)
+  }
+
+  ## remove ~ if it was added to the beginning of the path (like ~/foo/bar/baz instead of foo/bar/baz)
+  if (path_pieces[1] == "~") {
+    path_pieces <- path_pieces[-1]
+  }
+
```

Comment:
I suspect it would make more sense to deal with (most of?) this in the caller, i.e. checking and preparing `path`. This check + prep is a good candidate for a helper function, because I bet it comes up elsewhere.